app/lib/katello/resources/candlepin/product.rb

```diff
@@ -72,8 +72,13 @@ def remove_content(owner_label, product_id, content_id)
 
           def create_unlimited_subscription(owner_key, product_id, start_date)
             start_date ||= Time.now
-            # End it 100 years from now
-            end_date ||= start_date + 10_950.days
+
+            # Subscription-manager (python-rhsm) can't read the certificate with end date deyond
+            # 2049 year correctly. Refer the links below for more details:
+            # https://bugzilla.redhat.com/show_bug.cgi?id=1789654
```

Comment:
Most of the comments are private, such as linking to RFC 5280 that states:
> CAs conforming to this profile MUST always encode certificate validity dates through the year 2049 as UTCTime; certificate validity dates in 2050 or later MUST be encoded as GeneralizedTime.

Reply:
i can make that comment not-private